metrics/metrics.go

```diff
@@ -84,21 +83,20 @@ func RegisterRootScope(rep ScopeFunc) {
 		panic("There can be only one metrics reporter")
 	}
 
-	_scopeFunc = rep
+	_scopeFunc = scopeFunc
 }
 
-// RootScope returns the provided stats reporter, or nil
-func RootScope(i ScopeInit) (tally.Scope, io.Closer) {
+// RootScope returns the provided metrics scope and stats reporter, or nil if not provided
+func RootScope(i ScopeInit) (tally.Scope, tally.StatsReporter, io.Closer) {
 	_mu.Lock()
 	defer _mu.Unlock()
 
 	if _scopeFunc != nil {
-		rep, closer, err := _scopeFunc(i)
+		scope, reporter, closer, err := _scopeFunc(i)
 		if err != nil {
 			panic(fmt.Sprintf("Failed to initialize metrics reporter %v", err))
 		}
-		return rep, closer
+		return scope, reporter, closer
 	}
-
-	return nil, nil
+	return nil, nil, nil
```

Comment:
What makes it strange? In other words, what's the advantage of wrapping this in a struct? We treat the error as a panic here, why would we want to return the error?